### ISSUE TITLE
Interactive Tutorial - Creating a Cluster

### ISSUE THREAD

**Issue by HBDUDE** [kubernetes/website#25061]
Needs a dark mode for enhanced accessibility.

**Comment by sftim** [comment]
/retitle Allow switching website between “light mode” and  “dark mode”

/kind feature
/area web-development
/priority awaiting-more-evidence

**Comment by sftim** [comment]
@HBDUDE interesting idea. If this gets a few :+1: reactions we'll be happy to accept it as a feature for the backlog.

**Comment by zaheeruddinfaiz** [comment]
It would be great to have such an option. Since reading the documentation, the dark mode helps in reading for longer period of time.

**Comment by stevek-pro** [comment]
It would save energy, too.

**Comment by mhrbmr** [comment]
Just in curiosity, because I’m new to open source software, but does that mean anyone can potentially contribute? Say if someone wanted to build a dark mode out and then submitted it for review, would that be something someone could do?

**Comment by stevek-pro** [comment]
Indeed, you can and it will be much appreciated. I already started working on it. If you want to coordinate and help, let me know.

**Comment by mhrbmr** [comment]
Ok, that’s really cool to know. Im not extremely experienced as a dev yet, and so I don’t know if I’d be of much use, but I’d love to check out the code and see how it goes.

**Comment by vickyvikas7988** [comment]
+1 
Need dark mode for documentation

**Comment by stevek-pro** [comment]
Please someone provide location of central SASS (or similar) file. I asked in Slack, w/o result.

**Comment by sftim** [comment]
@stevek-pro this is a backlog priority issue. SIG Docs has a really chunky amount of work that we've triaged as more valuable, so it's likely that this will move forward specifically when someone who wants to see this specific improvement land picks it up.

If we don't see any movement within a few months, we have a bot that would close the feature request.

**Comment by sftim** [comment]
See https://stuffandnonsense.co.uk/blog/redesigning-your-product-and-website-for-dark-mode for a CSS approach towards implementing this.

**Comment by sftim** [comment]
Any implementation of this needs to work for diagrams and other graphics, as well as text. We don't want to have people who prefer dark mode missing out on being able to see what a diagram says.

**Comment by stevek-pro** [comment]
Sorry I did not follow up earlier, but since I discovered dark reader plugin, I am in no need of this feature anymore.

**Comment by sftim** [comment]
As well as using CSS, if we find we need JavaScript for some aspect then that's also feasible, eg https://flaviocopes.com/javascript-detect-dark-mode/

**Comment by sftim** [comment]
Duplicated by #39263

**Comment by KhushPatibandha** [comment]
I don't see any dark mode button on the kubernetes website, so i suppose this issue is still not resolved? so can I work on this? I think I can help.
Do let me know.

**Comment by KhushPatibandha** [comment]
can someone help me locate the main css file which is relevant to make a change in?

**Comment by sftim** [comment]
> can someone help me locate the main css file which is relevant to make a change in?

[`assets/scss/_base.scss`](https://github.com/kubernetes/website/blob/main/assets/scss/_base.scss)

This is SCSS, not CSS.

Try selecting `body.td-home` (or children), and using the appropriate media query.

**Comment by Gauravpadam** [comment]
Hey @KhushPatibandha, I see that you're working on this feature. Would you like to collaborate on a PR?

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/pull/43084

**Comment by sftim** [comment]
Duplicated by https://github.com/kubernetes/website/issues/44159

**Comment by sftim** [comment]
Help is welcome. Please consider working on #37444 first.

**Comment by dipesh-rawat** [comment]
Another duplicate request https://github.com/kubernetes/website/issues/44323

**Comment by sftim** [comment]
Help is welcome. Please consider working on #37444 first.

**Comment by dipesh-rawat** [comment]
In the long term, we might want to consider aligning with [Docsy](https://www.docsy.dev/) since it now supports **[Dark Mode (announcement here)](https://www.docsy.dev/blog/2024/0.10.0/#color-themes-and-dark-mode-support)**.

**Comment by sftim** [comment]
We have plans to upgrade to a newer Docsy; even if we do, we'll still need to adapt this website specifically to support both light and dark views.

**Comment by sftim** [comment]
Duplicated by https://github.com/kubernetes/website/issues/49433

**Comment by sftim** [comment]
I suggest we close this issue after April 2025 if https://github.com/kubernetes/website/issues/37444 has not yet been implemented. There doesn't seem to be a lot of interest in staffing the work.

**Comment by milinddethe15** [comment]
I want to take up this issue.
I will start up by:
* upgrading the docsy version to  v0.10 which supports [Light/dark mode menu](https://www.docsy.dev/blog/2024/0.10.0/#color-themes-and-dark-mode-support)
* And from there will see what extra changes/efforts are required.

**Comment by sftim** [comment]
@milinddethe15 noted, but we should work on https://github.com/kubernetes/website/issues/37444

You can review that pull request if you'd like to.

**Comment by sftim** [comment]
Good news, everyone: #37444 is done. Now we just have all the rest of the site to do&hellip;

**Comment by Alex-Caesar** [comment]
Great work so far guys!

**Comment by lmktfy** [comment]
Help is still welcome; so far, we only made this work with one page.

Covering more of the site (even if only a subset of all the sections) would be great.

**Comment by lmktfy** [comment]
Duplicated by https://github.com/kubernetes/website/issues/52358

**Comment by ishaanxgupta** [comment]
Hi @lmktfy I had raised a duplicate issue for this, the home page still lags a toggle button and also not all pages have a dark mode. Should I raise a PR for this?

**Comment by lmktfy** [comment]
> the home page still lags a toggle button and also not all pages have a dark mode. Should I raise a PR for this?

My opinion: many small PRs will be more likely to be accepted then a single PR that does all the required changes. Contributors picking this up need to know how to split the outstanding work so that each accepted PR represents an incremental improvement.

---

Consider adding feature flags. For example, a toggle UI that only appears (or works) if you have opted in to the associated experiment. Bear in mind that this issue is already 5 years old and is **not** a project priority; we can be confident that progress will be slow.

**Comment by ishaanxgupta** [comment]
Yes will do small PRs, could you please assign me on this one?
Also like a small question that if this is not a priority issue, then what other goals are we looking for in this website module?

**Comment by lmktfy** [comment]
> if this is not a priority issue, then what other goals are we looking for in this website module?

See [important-soon](https://github.com/kubernetes/website/issues?q=is%3Aissue%20state%3Aopen%20label%3Apriority%2Fimportant-soon) and [important-longterm](https://github.com/kubernetes/website/issues?q=is%3Aissue%20state%3Aopen%20label%3Apriority%2Fimportant-longterm%20sort%3Acreated-asc).
